src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -18,6 +18,8 @@ public class PhysicalFileProvider : IFileProvider, IDisposable
         private const string PollingEnvironmentKey = "DOTNET_USE_POLLING_FILE_WATCHER";
         private static readonly char[] _invalidFileNameChars = Path.GetInvalidFileNameChars()
             .Where(c => c != Path.DirectorySeparatorChar && c != Path.AltDirectorySeparatorChar).ToArray();
+        private static readonly char[] _invalidFilterChars = _invalidFileNameChars
```

Comment:
why are we doing this? it does not look related to the leading slashes bug your fixing.